import "soc_status.proto";
import "position_status.proto";
import "vehicle_status.proto";

option java_package = "com.nextev.cvs_proto.protobuf.events";
option java_outer_classname = "JourneyEndMessage";

message JourneyEndEvent {

    optional string id = 1; // unique id, VIN for vehicle data
    optional int32 version = 2; // message protocol version
    optional string journey_id = 3; // JourneyID = YYMMDD####
    optional uint64 sample_ts = 4; // sample timestamp in milliseconds
    optional PositionStatus position_status = 5;
    optional SOCStatus soc_status = 6; // soc status
    optional VehicleStatus vehicle_status = 7;
}